test/TestCase.h

```diff
@@ -31,6 +31,43 @@
 namespace solidity::frontend::test
 {
 
+/**
+ * A reader for test case data file, which parses source, settings and (optionally) simple expectations.
+ */
+class TestCaseReader
+{
+public:
+	TestCaseReader() = default;
+	TestCaseReader(std::string const& _filename);
+
+	std::map<std::string, std::string> sources() { return m_sources; }
+	std::string source();
+	std::size_t lineNumbers() { return m_lineNumbers; }
+	std::map<std::string, std::string> settings() { return m_settings; }
```

Comment:
Why not return by `const&`?

Reply:
I was not sure if we should consider a theoretical future case when `TestCaseReader::settings()` is used somewhere else, and the lifetime of `TestCaseReader` is not long enough. Anyway, changed to `const&`, 6c9736b .